Unittest for the autodetection fails when running on a macbook with the M1 apple chip

The unittest for the autodetection fails when running on a macbook with the M1 apple chip:
```
================================================================== FAILURES ===================================================================
________________________________________________________________ test_autotect ________________________________________________________________

default_exec_ctx = <unittests.conftest._ExecutionContext object at 0x100852d60>

    def test_autotect(default_exec_ctx):
        autodetect.detect_topology()
        part = runtime().system.partitions[0]
        assert part.processor.info == cpuinfo()
        if part.processor.info:
>           assert part.processor.num_cpus == part.processor.info['num_cpus']
E           KeyError: 'num_cpus'

unittests/test_autodetect.py:77: KeyError
------------------------------------------------------------ Captured stderr call -------------------------------------------------------------
sysctl: unknown oid 'machdep.cpu.leaf7_features'
sysctl: unknown oid 'machdep.cpu.vendor'
sysctl: unknown oid 'machdep.cpu.model'
sysctl: unknown oid 'machdep.cpu.leaf7_features'
sysctl: unknown oid 'machdep.cpu.vendor'
sysctl: unknown oid 'machdep.cpu.model'
=========================================================== short test summary info ===========================================================
FAILED unittests/test_autodetect.py::test_autotect - KeyError: 'num_cpus'
```

The command `sysctl hw machdep.cpu.cache`, which is used in the `cpuinfo` gives an error: 
```
sysctl: unknown oid 'machdep.cpu.cache'
```

--- a/reframe/utility/cpuinfo.py
+++ b/reframe/utility/cpuinfo.py
@@ -11,6 +11,7 @@ import re
 
 import reframe.utility.osext as osext
 from reframe.core.exceptions import SpawnedProcessError
+from reframe.core.logging import getlogger
 
 
 def _bits_from_str(mask_s):
@@ -206,32 +207,44 @@ def _sysctl_topo():
     match = re.search(r'hw\.ncpu: (?P<num_cpus>\d+)', exec_output.stdout)
     if match:
         num_cpus = int(match.group('num_cpus'))
+    else:
+        getlogger().warning(f'not able to detect num_cpus')
 
     match = re.search(r'hw\.physicalcpu: (?P<num_cores>\d+)',
                       exec_output.stdout)
     if match:
         num_cores = int(match.group('num_cores'))
+    else:
+        getlogger().warning(f'not able to detect num_cores')
 
     match = re.search(r'hw\.packages: (?P<num_sockets>\d+)',
                       exec_output.stdout)
     if match:
         num_sockets = int(match.group('num_sockets'))
         cpuinfo['num_sockets'] = num_sockets
+    else:
+        getlogger().warning(f'not able to detect num_sockets')
 
     match = re.search(r'hw\.cacheconfig:(?P<cacheconfig>(\s\d+)*)',
                       exec_output.stdout)
     if match:
         cacheconfig = list(map(int, match.group('cacheconfig').split()))
+    else:
+        getlogger().warning(f'not able to detect cacheconfig')
 
     match = re.search(r'hw\.cachesize:(?P<cachesize>(\s\d+)*)',
                       exec_output.stdout)
     if match:
         cachesize = list(map(int, match.group('cachesize').split()))
+    else:
+        getlogger().warning(f'not able to detect cachesize')
 
     match = re.search(r'hw\.cachelinesize: (?P<linesize>\d+)',
                       exec_output.stdout)
     if match:
         linesize = int(match.group('linesize'))
+    else:
+        getlogger().warning(f'not able to detect cache linesize')
 
     # index 0 is referring to memory
     cache_associativity = [0]
@@ -242,7 +255,13 @@ def _sysctl_topo():
         match = re.search(rf'machdep\.cpu\.cache\.L{i}_associativity: '
                           rf'(?P<associativity>\d+)',
                           exec_output.stdout)
-        assoc = int(match.group('associativity')) if match else 0
+        if match:
+            assoc = int(match.group('associativity'))
+        else:
+            assoc = 0
+            getlogger().warning(f'not able to detect L{i} cache '
+                                f'associativity')
+
         cache_associativity.append(assoc)
 
     num_cpus_per_socket = num_cpus // num_sockets

--- a/reframe/utility/cpuinfo.py
+++ b/reframe/utility/cpuinfo.py
@@ -11,7 +11,6 @@ import re
 
 import reframe.utility.osext as osext
 from reframe.core.exceptions import SpawnedProcessError
-from reframe.core.logging import getlogger
 
 
 def _bits_from_str(mask_s):
@@ -207,44 +206,32 @@ def _sysctl_topo():
     match = re.search(r'hw\.ncpu: (?P<num_cpus>\d+)', exec_output.stdout)
     if match:
         num_cpus = int(match.group('num_cpus'))
-    else:
-        getlogger().warning(f'not able to detect num_cpus')
 
     match = re.search(r'hw\.physicalcpu: (?P<num_cores>\d+)',
                       exec_output.stdout)
     if match:
         num_cores = int(match.group('num_cores'))
-    else:
-        getlogger().warning(f'not able to detect num_cores')
 
     match = re.search(r'hw\.packages: (?P<num_sockets>\d+)',
                       exec_output.stdout)
     if match:
         num_sockets = int(match.group('num_sockets'))
         cpuinfo['num_sockets'] = num_sockets
-    else:
-        getlogger().warning(f'not able to detect num_sockets')
 
     match = re.search(r'hw\.cacheconfig:(?P<cacheconfig>(\s\d+)*)',
                       exec_output.stdout)
     if match:
         cacheconfig = list(map(int, match.group('cacheconfig').split()))
-    else:
-        getlogger().warning(f'not able to detect cacheconfig')
 
     match = re.search(r'hw\.cachesize:(?P<cachesize>(\s\d+)*)',
                       exec_output.stdout)
     if match:
         cachesize = list(map(int, match.group('cachesize').split()))
-    else:
-        getlogger().warning(f'not able to detect cachesize')
 
     match = re.search(r'hw\.cachelinesize: (?P<linesize>\d+)',
                       exec_output.stdout)
     if match:
         linesize = int(match.group('linesize'))
-    else:
-        getlogger().warning(f'not able to detect cache linesize')
 
     # index 0 is referring to memory
     cache_associativity = [0]
@@ -255,13 +242,7 @@ def _sysctl_topo():
         match = re.search(rf'machdep\.cpu\.cache\.L{i}_associativity: '
                           rf'(?P<associativity>\d+)',
                           exec_output.stdout)
-        if match:
-            assoc = int(match.group('associativity'))
-        else:
-            assoc = 0
-            getlogger().warning(f'not able to detect L{i} cache '
-                                f'associativity')
-
+        assoc = int(match.group('associativity')) if match else 0
         cache_associativity.append(assoc)
 
     num_cpus_per_socket = num_cpus // num_sockets

--- a/reframe/utility/cpuinfo.py
+++ b/reframe/utility/cpuinfo.py
@@ -195,7 +195,7 @@ def _sysfs_topo():
 
 def _sysctl_topo():
     try:
-        exec_output = osext.run_command('sysctl hw machdep.cpu.cache',
+        exec_output = osext.run_command('sysctl hw machdep.cpu',
                                         check=True)
     except (FileNotFoundError, SpawnedProcessError):
         return {}

